ci/test/check-merge-with-target.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# check-merge-with-target.sh — check if merge with target branch succeeds and checks still pass.
+
+set -euo pipefail
+
+. misc/shlib/shlib.bash
+
+ci_try git fetch "$BUILDKITE_REPO"
```

Comment:
I think ci_try is wrong here, since we want to abort if the `git fetch` (and other commands) fail, not continue.

Reply:
✅ Removed `ci_try` everywhere.